File: src/main/java/io/github/binaryfoo/decoders/bit/BitStringField.java
package io.github.binaryfoo.decoders.bit;

import io.github.binaryfoo.bit.EmvBit;

import java.util.Set;

public interface BitStringField {
    String getPositionIn(Set<EmvBit> bits);
    String getValueIn(Set<EmvBit> bits);
    int getStartBytesOffset();
    int getLengthInBytes();
}

File: src/main/java/io/github/binaryfoo/decoders/bit/BitStringField.kt
package io.github.binaryfoo.decoders.bit

import io.github.binaryfoo.bit.EmvBit

interface BitStringField {
    fun getPositionIn(bits: Set<EmvBit>?): String
    fun getValueIn(bits: Set<EmvBit>): String?
    fun getStartBytesOffset(): Int
    fun getLengthInBytes(): Int
}
